packages/core/src/components/templates/SocialNetworkLinks/SocialNetworkLinks.tsx

```diff
@@ -16,56 +16,83 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 import {StyleSheet, View} from 'react-native';
 import {Icon, useThemeColor} from '@axelor/aos-mobile-ui';
 import {linkingProvider} from '../../../tools';
 
-const formatGoogleSearch = (name, lastName, company) => {
+const formatGoogleSearch = (name, lastName, company, linkedinEnterprise) => {
   let result = '';
 
   if (name != null && lastName != null) {
     result = result.concat(name).concat('+').concat(lastName);
   }
 
-  if (company != null) {
+  if (company != null && !linkedinEnterprise) {
     result = result.concat('+').concat(company).concat('+');
   }
 
   return result;
 };
 
-const formatLinkedinSearch = (name, lastName, company) => {
-  if (name != null && lastName != null) {
-    return `${name}/${lastName}`;
+const formatLinkedinSearch = (name, lastName, company, linkedinEnterprise) => {
+  if (company != null && linkedinEnterprise) {
+    return `company/${company}`;
   }
-
-  if (company != null) {
-    return company;
+  if (name != null && lastName != null) {
+    return `pub/dir/${name}/${lastName}`;
   }
 
   return '';
 };
 
+const getFirstNameAndName = nameToSplit => {
+  const fullName = nameToSplit?.split(' ');
+  if (fullName?.length === 2) {
+    return {firstName: fullName[0], lastName: fullName[1]};
+  } else {
+    return {firstName: fullName, lastName: ''};
+  }
+};
+
 const SocialNetworkLinks = ({
+  style,
   size = 20,
   data,
   googleColor,
   hideGoogle = false,
   linkedinColor,
   hideLinkedin = false,
+  linkedinEnterprise = false,
 }: {
+  style?: any;
   size?: number;
-  data: {name?: string; lastName?: string; company?: string};
+  data: {
+    fullName?: string;
+    name?: string;
+    lastName?: string;
+    company?: string;
+  };
   googleColor?: string;
   hideGoogle?: boolean;
   linkedinColor?: string;
   hideLinkedin?: boolean;
+  linkedinEnterprise?: boolean;
 }) => {
   const Colors = useThemeColor();
+  const [objectName, setObjectName] = useState({
+    firstName: data?.name,
+    lastName: data?.lastName,
+  });
+
+  useEffect(() => {
+    if (data?.fullName != null) {
+      setObjectName(getFirstNameAndName(data?.fullName));
+    }
+  }, [data]);
```

Comment:
Do not use state here : 
1. You are trigerring refresh only on fullName, name and lastName are never updated...
2. This is complexifying the component when it's not needed.

Give full data to both formatter and split fullName if needed in the formatter.